### PR TITLE
[NTUSER] Improve WINSTATION_OBJECT compatibility

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-18252](https://jira.reactos.org/browse/CORE-18252)

## Proposed changes

- Add a comment for `WINSTATION_OBJECT` structure.
- Modify `WINSTATION_OBJECT` structure with inserting `struct tagKL *spklList` member.
- Add some `C_ASSERT(offsetof(WINSTATION_OBJECT, ...) == ...);`.

### CODE REVIEW

File: win32ss/user/ntuser/winsta.h [hunk 1]
```diff
@@ -11,18 +11,16 @@
...
     LIST_ENTRY DesktopListHead;
     PRTL_ATOM_TABLE AtomTable;
-    HANDLE ShellWindow;
-    HANDLE ShellListView;
-
-    ULONG Flags;
-    struct _DESKTOP* ActiveDesktop;
 
+    ULONG          Flags;
+    struct tagKL  *spklList;
```
Comment (HBelusca) [inline win32ss/user/ntuser/winsta.h:23]:
```suggestion
    struct tagKL*  spklList;
```

Comment (HBelusca) [reply win32ss/user/ntuser/winsta.h:23]: We cannot use "PKL" as type directly here?

File: win32ss/user/ntuser/winsta.h [hunk 2]
```diff
@@ -40,8 +38,21 @@ typedef struct _WINSTATION_OBJECT
     LUID           luidUser;
     PVOID          psidUser;
 
+    struct _DESKTOP* ActiveDesktop;
```
Comment (HBelusca) [inline win32ss/user/ntuser/winsta.h:41]:
```suggestion
    /* ReactOS-specific */
    struct _DESKTOP* ActiveDesktop;
```

File: win32ss/user/ntuser/winsta.h [hunk 3]
```diff
@@ -40,8 +38,21 @@ typedef struct _WINSTATION_OBJECT
...
+    HANDLE         ShellWindow;
+    HANDLE         ShellListView;
 } WINSTATION_OBJECT, *PWINSTATION_OBJECT;
 
+#ifndef _WIN64
```
Comment (HBelusca) [inline win32ss/user/ntuser/winsta.h:46]: As said in the JIRA report, this structure is internal to win32k only so it's not mandatory to have the very same layout (especially when fields are different due to fundamental differences in implementations).

File: win32ss/user/ntuser/winsta.h (hunk 1 above)
Comment (katahiromz) [reply win32ss/user/ntuser/winsta.h:23]: No we can't.

Comment (HBelusca) [reply win32ss/user/ntuser/winsta.h]: OK that's certainly like some of these other structures (like desktop etc) being used there.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**